### PR TITLE
chore(VSCode): Refactor KaotoEditor structure

### PR DESCRIPTION
**Pull request by lordrip**
### Context
Currently, the `DesignPage` component is duplicated to serve the multiplying architecture and the webapp.

In addition to that, for the multiplying architecture, we're using a React render approach to navigate between different tabs, whereas in the web app we use a Router.

This commit removes the `Tabs` folder and uses the `DesignPage` component directly to reduce the duplication.

In addition to that, a Router has been introduced to navigate between the tabs, this is a cleanup for the upcoming tasks.

### CODE REVIEW

File: packages/ui/src/multiplying-architecture/KaotoEditorRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        lazy: async () => import('../pages/Beans'),
+      },
+      {
+        path: Links.Metadata,
+        lazy: async () => import('../pages/Metadata'),
+      },
+      {
+        path: Links.PipeErrorHandler,
+        lazy: async () => import('../pages/PipeErrorHandler'),
+      },
```
Comment (lordrip) [inline packages/ui/src/multiplying-architecture/KaotoEditorRouter.tsx:27]: @apupier we could add an `About` tab here.

File: packages/ui/src/multiplying-architecture/KaotoEditorApp.tsx [hunk 2]
```diff
@@ -30,19 +31,6 @@ export class KaotoEditorApp implements Editor {
     this.editorRef = createRef<EditorApi>();
   }
 
-  async getElementPosition() {
```
Comment (lordrip) [inline packages/ui/src/multiplying-architecture/KaotoEditorApp.tsx:33]: This is no longer used.

File: packages/ui/src/multiplying-architecture/KaotoEditor.tsx [hunk 3]
```diff
@@ -59,87 +55,91 @@ export const KaotoEditor = () => {
...
-                  <Icon>
-                    <img src={camelIcon} alt="Camel icon" />
-                  </Icon>
-                </TabTitleIcon>
-                <TabTitleText>Design</TabTitleText>
-              </>
-            }
-            aria-label="Design canvas"
-          >
-            <DesignTab />
```
Comment (lordrip) [inline packages/ui/src/multiplying-architecture/KaotoEditor.tsx:83]: No more linking components directly, the router will take care of mounting the right component when needed.

File: packages/ui/src/multiplying-architecture/KaotoBridge.tsx [hunk 4]
```diff
@@ -147,11 +144,7 @@ export const KaotoBridge = forwardRef<EditorApi, PropsWithChildren<KaotoBridgePr
         <SchemasLoaderProvider>
           <CatalogLoaderProvider>
             <CatalogTilesProvider>
-              <VisibleFlowsProvider>
-                <CatalogModalProvider>
-                  <DeleteModalContextProvider>{props.children}</DeleteModalContextProvider>
-                </CatalogModalProvider>
```
Comment (lordrip) [inline packages/ui/src/multiplying-architecture/KaotoBridge.tsx:153]: These providers already come directly from the `DesignPage`, so they are no longer needed here.

**Review by apupier — APPROVED** [review]

**Pull request merged**